pkg/sql/hints/hints_test: TestHintCacheMultiNode failed

pkg/sql/hints/hints_test.TestHintCacheMultiNode [failed](https://mesolite.cluster.engflow.com/invocations/default/c5b198f2-c908-4d61-ab47-f607d4665375?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2hpbnRzOmhpbnRzX3Rlc3Q=) on release-25.4.4-rc @ [3b602e9ede0a2153e9d0ebd5823e9d1444e0d4fc](https://github.com/cockroachdb/cockroach/commits/3b602e9ede0a2153e9d0ebd5823e9d1444e0d4fc):


```
=== RUN   TestHintCacheMultiNode
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestHintCacheMultiNode296488745
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    hint_cache_test.go:273: condition failed to evaluate within 3m45s: from hint_cache_test.go:506: expected hash 6162414449535352931 with hasHints=false, got hasHints=true
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestHintCacheMultiNode296488745
--- FAIL: TestHintCacheMultiNode (541.81s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #161857 pkg/sql/hints/hints_test: TestHintCacheMultiNode failed [C-investigation C-test-failure O-robot T-sql-queries branch-release-26.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestHintCacheMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59373

--- a/pkg/base/test_server_args.go
+++ b/pkg/base/test_server_args.go
@@ -435,6 +435,11 @@ var (
 	// worth the cost of never running that test with the virtualization
 	// layer active.
 	TestNeedsTightIntegrationBetweenAPIsAndTestingKnobs = TestIsSpecificToStorageLayerAndNeedsASystemTenant
+
+	// TestSkipSecondaryTenantsUnderDuress should be used whenever we want to
+	// disable test tenant randomization under heavy configs (e.g. under race)
+	// due to overload.
+	TestSkipSecondaryTenantsUnderDuress = TestIsSpecificToStorageLayerAndNeedsASystemTenant
 )
 
 func (do DefaultTestTenantOptions) AllowAdditionalTenants() bool {

--- a/pkg/sql/hints/hint_cache_test.go
+++ b/pkg/sql/hints/hint_cache_test.go
@@ -269,10 +269,10 @@ func TestHintCacheMultiNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	// Skip external mode when running under race to avoid flakes.
+	// Skip secondary tenants when running under race to avoid flakes.
 	var serverArgs base.TestServerArgs
 	if util.RaceEnabled {
-		serverArgs.DefaultTestTenant = base.TestSkippedForExternalModeDueToPerformance(161857)
+		serverArgs.DefaultTestTenant = base.TestSkipSecondaryTenantsUnderDuress
 	}
 
 	ctx := context.Background()

